python/lsst/ip/isr/isrFunctions.py

```diff
@@ -778,17 +781,32 @@ def gainContext(exp, image, apply):
         Image to apply/remove gain.
     apply : `Bool`
         If True, apply and remove the amplifier gain.
+    gains : `dict` of `float`
+        A dictionary, keyed by amplifier name, of the gains to use.
+        If gains is None, the nominal gains in the amplifier object are used.
 
     Yields
     ------
     exp : `lsst.afw.image.Exposure`
         Exposure with the gain applied.
     """
+    # check we have all of them if provided because mixing and matching would
+    # be a real mess
+    if gains:
```

Comment:
Should this check if that both `apply is True and gains`?  That might be excessively pedantic.

Reply:
Yeah, fair enough, I can't really see a downside to that.